From 8a293f06120f3026fa9db913b4339e12e15b76b3 Mon Sep 17 00:00:00 2001
From: Dan Walkes <danwalkes@trellis-logic.com>
Date: Sun, 23 Feb 2025 08:38:53 -0700
Subject: [PATCH] configs: reduce CONFIG_ENV_SIZE on t186

Targets p2771 (tx2) and p3636 (tx2-nx) have a conflict in
uboot environment storage location and bootinfo storage
with tegra-bootinfo tool.

As an alternative to resizing the extended sector count
on tegra-bootinfo, this change reduces uboot environment
storage size and shifts the redundant storage area
out of the area used by tegra-bootinfo (starting at
0x3f7800) to avoid overwriting the tegra-bootinfo
storage location.  I
---
 configs/p2771-0000-000_defconfig | 4 ++--
 configs/p2771-0000-500_defconfig | 4 ++--
 configs/p3636-0001_defconfig     | 6 ++----
 3 files changed, 6 insertions(+), 8 deletions(-)

diff --git a/configs/p2771-0000-000_defconfig b/configs/p2771-0000-000_defconfig
index 21ec2a383a..c81f4bde6e 100644
--- a/configs/p2771-0000-000_defconfig
+++ b/configs/p2771-0000-000_defconfig
@@ -39,9 +39,9 @@ CONFIG_BOOTP_PREFER_SERVERIP=y
 CONFIG_CMD_CACHE=y
 CONFIG_NVME=y
 CONFIG_OF_LIBFDT_OVERLAY=y
-CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_SIZE=0x1C000
 CONFIG_ENV_OFFSET=0x3bb000
-CONFIG_ENV_OFFSET_REDUND=0x3db000
+CONFIG_ENV_OFFSET_REDUND=0x3d7000
 CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
 # CONFIG_ENV_IS_NOWHERE is not set
 # CONFIG_ENV_IS_IN_EEPROM is not set
diff --git a/configs/p2771-0000-500_defconfig b/configs/p2771-0000-500_defconfig
index d37011c070..721d5ea0ee 100644
--- a/configs/p2771-0000-500_defconfig
+++ b/configs/p2771-0000-500_defconfig
@@ -39,9 +39,9 @@ CONFIG_BOOTP_PREFER_SERVERIP=y
 CONFIG_CMD_CACHE=y
 CONFIG_NVME=y
 CONFIG_OF_LIBFDT_OVERLAY=y
-CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_SIZE=0x1C000
 CONFIG_ENV_OFFSET=0x3bb000
-CONFIG_ENV_OFFSET_REDUND=0x3db000
+CONFIG_ENV_OFFSET_REDUND=0x3d7000
 CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
 # CONFIG_ENV_IS_NOWHERE is not set
 # CONFIG_ENV_IS_IN_EEPROM is not set
diff --git a/configs/p3636-0001_defconfig b/configs/p3636-0001_defconfig
index 4415ae80bb..ca4859bcd2 100644
--- a/configs/p3636-0001_defconfig
+++ b/configs/p3636-0001_defconfig
@@ -1,8 +1,6 @@
 CONFIG_ARM=y
 CONFIG_TEGRA=y
 CONFIG_SYS_TEXT_BASE=0x80080000
-CONFIG_ENV_SIZE=0x2000
-CONFIG_ENV_OFFSET=0xFFFD8000
 CONFIG_NR_DRAM_BANKS=1026
 CONFIG_TEGRA186=y
 CONFIG_TARGET_P3636_0001=y
@@ -42,9 +40,9 @@ CONFIG_BOOTP_PREFER_SERVERIP=y
 CONFIG_CMD_CACHE=y
 CONFIG_NVME=y
 CONFIG_OF_LIBFDT_OVERLAY=y
-CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_SIZE=0x1C000
 CONFIG_ENV_OFFSET=0x3bb000
-CONFIG_ENV_OFFSET_REDUND=0x3db000
+CONFIG_ENV_OFFSET_REDUND=0x3d7000
 CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
 # CONFIG_ENV_IS_NOWHERE is not set
 # CONFIG_ENV_IS_IN_EEPROM is not set
-- 
2.34.1

